src/radix_tree.cpp

```diff
@@ -517,35 +518,25 @@ bool zmq::radix_tree::check (const unsigned char *key, size_t size)
 
 static void
 visit_keys (node n,
-            unsigned char **buffer,
-            size_t buffer_size,
-            size_t maxbuffer_size,
+            std::vector<unsigned char> &buffer,
             void (*func) (unsigned char *data, size_t size, void *arg),
             void *arg)
 {
-    if (buffer_size >= maxbuffer_size) {
-        maxbuffer_size += 256;
-        *buffer =
-          static_cast<unsigned char *> (realloc (*buffer, maxbuffer_size));
-        zmq_assert (*buffer);
-    }
-
     for (size_t i = 0; i < n.prefix_length (); ++i)
-        (*buffer)[buffer_size++] = n.prefix ()[i];
+        buffer.push_back (n.prefix ()[i]);
     if (n.refcount () > 0)
-        func (*buffer, buffer_size, arg);
+        func (buffer.data (), buffer.size (), arg);
```

Comment:
1>..\libzmq\src\radix_tree.cpp(528) : error C2039: 'data' : is not a member of 'std::vector<_Ty>'
1>        with
1>        [
1>            _Ty=unsigned char
1>        ]
1>trie.cpp
1>thread.cpp

This breaks the build on Windows (see Appveyor) could you please check?

Reply:
Sorry, looks like `vector::data` was added in C++11. Also it doesn't necessarily return NULL when the vector is empty, which was also not correct. I've attempted a fix in the latest commit.